src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -109,17 +109,10 @@ static int process_resume(void) {
         return 0;
 }
 
-static int run(int argc, char *argv[]) {
+static int run(const char *dest, const char *dest_early, const char *dest_late) {
         int r = 0;
 
-        log_setup_generator();
-
-        if (argc > 1 && argc != 4)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "This program takes three or no arguments.");
-
-        if (argc > 1)
-                arg_dest = argv[1];
+        arg_dest = ASSERT_PTR(dest);
```

Comment:
isn't this a change in behaviour? there was a default before, now there isn't - at least the commit message should explain the change?

Reply:
maybe postpone those changes to the later commit?

Reply:
@bluca I think you might be confused by github's reordering of commits. The commit is one of the first ones, and as @yuwata wrote, `DEFINE_MAIN_GENERATOR_FUNCTION()` always sets `dest`.

Reply:
I added a comment in the commit message.